docs/architecture.rst

```diff
@@ -1,142 +1,243 @@
-
-* **Secrets API**: Provides a simple interface for storing and retrieving secrets to a allow a more seamless
-  experience when accessing resources across environments. It provides a simple interface for storing and retrieving
-  secrets from a variety of providers (e.g. AWS, Azure, GCP, Hugging Face, Github, etc.) as well as SSH Keys and
-  custom secrets, and stores them in Hashicorp Vault.
-
-* **Configs**: Set and preserve default configs across environments.
-
-Management
-----------
-
-Runhouse provides tools for visibility and management of resources as long-living assets shared by teams or projects.
-
-* `Runhouse Den <https://www.run.house/dashboard>`_: provides an individual or admin view of all resources, secrets,
-  groups, and sharing. Resource metadata is automatically versioned in RNS, allowing teams to maintain single-sources
-  of truth for assets with zero downtime to update or roll back, and trace exact lineage for any existing resource.
-
-* **Organization structure**: Both resources and users can be organized into organizations to apply and manage shared
-  access and permissions. Resources can be created for an organization by specifying its name in full
-  when creating or loading down a Runhouse resource.
-
-  .. code-block:: python
-
-   import runhouse as rh
-
-   # Create a new function for the organization named "my_org_name"
-   remote_func = rh.function(name="/my_org_name/my_function",
-                             fn=my_func).to(system=my_cluster, env=my_env)
-
-   # Reload the function and invoke it remotely on the cluster
-   remote_func = rh.function(name="/my_org_name/my_function")
-   res = remote_func(*args, **kwargs)
-
-  .. note::
-
-    You must be an organization admin in order to add new members and manage their permissions within the organization.
-    The management panel can be accessed in the `Den account page <https://www.run.house/account#organizations>`_.
-
-Resource Access Levels
-----------------------
-
-Runhouse lets you control the access levels you provide to the users of your resources:
-
-- :code:`read` (default): Users cannot modify the config, but can access the resource directly.
-- :code:`write`: Users have full control over the resource, including modifying its config.
-
-
-Resource Visibility
--------------------
+    import runhouse as rh
+
+    # [1] and [2]
+    gpu = rh.cluster(name="rh-a10x", instance_type="A10G:1", provider="aws").up_if_not()
+
+    # [3]
+    sd_env = rh.env(reqs=["torch", "transformers", "diffusers"], name="sd_generate")
+    remote_sd_generate = rh.function(sd_generate).to(gpu, worker=sd_env)
+
+    # [4]
+    imgs = remote_sd_generate("A hot dog made out of matcha.")
+    imgs[0].show()
+
+    # [5]
+    remote_sd_generate.save()
+    sd_upsampler = rh.function(name="/my_username/sd_upsampler")
+    high_res_imgs = sd_upsampler(imgs)
+
+    # [6]
+    gpu.teardown()
+
+.. note::
+
+
+1. Specify and/or Allocate Compute
```

Comment:
the length of these explanations is long enough that one would need to scroll back and forth between the code snippets. thoughts on rewriting the code snippet corresponding to the explanation below each number too?



**#. header**
```
gpu = rh.cluster(name="rh-a10x", instance_type="A10G:1", provider="aws").up_if_not()
```
< desc >